### ISSUE TITLE
How can I see F1-score and precision/recall curves for validation set ?

### ISSUE THREAD

**Issue by Grhanas** [ultralytics/yolov5#13348]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello! I can see the F1-score for the training dataset after completing the training process. Additionally, I have access to the dfl_loss, box_loss, and cls_loss for both the training and validation sets, as shown in "results.png." However, I'm unable to find precision and recall values for the validation set. I would like to understand if my model is experiencing overfitting or underfitting.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Grhanas, thank you for reaching out with your question! 🚀

To get started with understanding your validation metrics, you might want to check out our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) where we cover various aspects including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and more.

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us investigate further.

For training-related questions, ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Including more details like dataset examples and full training logs can be very helpful.

## Requirements

Ensure you are using [**Python>=3.8.0**](https://www.python.org/) with all dependencies from [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt), including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). Here's a quick setup guide:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 can be run in various environments. Some options include:

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Learn more in the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

Current CI status: <a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a> If this badge is green, all tests are passing.

## Discover YOLOv8 🚀

Explore the capabilities of our latest model [YOLOv8](https://github.com/ultralytics/ultralytics) for even better performance in object detection, image segmentation, and classification. Check the [YOLOv8 Docs](https://docs.ultralytics.com/) and get started with this command:
```bash
pip install ultralytics
```

This is an automated response, but an Ultralytics engineer will review and assist you soon. Thank you for your patience! 😊

**Comment by pderrenger** [comment]
@Grhanas to view precision, recall, and F1-score curves for the validation set, you can check the `results.png` file generated in the `runs/train/exp` directory after training. This file includes precision-recall (PR) curves, which can help you assess model performance and overfitting.
